velox/dwio/dwrf/writer/WriterContext.cpp

```diff
@@ -138,10 +137,24 @@ int64_t WriterContext::availableMemoryReservation() const {
       generalPool_->availableReservation();
 }
 
-void WriterContext::releaseMemoryReservation() {
+int64_t WriterContext::releasableMemoryBytes() const {
+  return (dictionaryPool_->reservedBytes() +
+          outputStreamPool_->reservedBytes() + generalPool_->reservedBytes()) -
```

Comment:
Can we just use availableMemoryReservation? It is roughly the same? I think we probably only want to free from the writer if it has sufficient free memory bytes?

Reply:
discussed offline